plugin-server/functional_tests/session-recordings-blob-ingestion.test.ts

```diff
@@ -0,0 +1,150 @@
+import { GetObjectCommand, GetObjectCommandOutput, ListObjectsV2Command, S3Client } from '@aws-sdk/client-s3'
+import fs from 'fs'
+import { Consumer, Kafka, KafkaMessage, logLevel } from 'kafkajs'
+import * as zlib from 'zlib'
+
+import { defaultConfig } from '../src/config/config'
+import { compressToString } from '../src/main/ingestion-queues/session-recording/blob-ingester/utils'
+import { getObjectStorage } from '../src/main/services/object_storage'
+import { UUIDT } from '../src/utils/utils'
+import { capture, createOrganization, createTeam } from './api'
+import { waitForExpect } from './expectations'
+
+let kafka: Kafka
+let organizationId: string
+
+let dlq: KafkaMessage[]
+let dlqConsumer: Consumer
+
+let s3: S3Client
+
+function generateVeryLongString(length = 1025) {
+    return [...Array(length)].map(() => Math.random().toString(36)[2]).join('')
+}
+
+beforeAll(async () => {
+    kafka = new Kafka({ brokers: [defaultConfig.KAFKA_HOSTS], logLevel: logLevel.NOTHING })
+
+    dlq = []
+    dlqConsumer = kafka.consumer({ groupId: 'session_recording_events_test' })
+    await dlqConsumer.subscribe({ topic: 'session_recording_events_dlq' })
+    await dlqConsumer.run({
+        eachMessage: ({ message }) => {
+            dlq.push(message)
+            return Promise.resolve()
+        },
+    })
+
+    organizationId = await createOrganization()
+
+    const objectStorage = getObjectStorage({
+        OBJECT_STORAGE_ENDPOINT: defaultConfig.OBJECT_STORAGE_ENDPOINT,
+        OBJECT_STORAGE_REGION: defaultConfig.OBJECT_STORAGE_REGION,
+        OBJECT_STORAGE_ACCESS_KEY_ID: defaultConfig.OBJECT_STORAGE_ACCESS_KEY_ID,
+        OBJECT_STORAGE_SECRET_ACCESS_KEY: defaultConfig.OBJECT_STORAGE_SECRET_ACCESS_KEY,
+        OBJECT_STORAGE_ENABLED: defaultConfig.OBJECT_STORAGE_ENABLED,
+        OBJECT_STORAGE_BUCKET: defaultConfig.OBJECT_STORAGE_BUCKET,
+    })
+    if (!objectStorage) {
+        throw new Error('S3 not configured')
+    }
+    s3 = objectStorage.s3
+})
+
+afterAll(async () => {
+    await Promise.all([await dlqConsumer.disconnect()])
+})
+
+test.skip(`single recording event writes data to local tmp file`, async () => {
+    const teamId = await createTeam(organizationId)
+    const distinctId = new UUIDT().toString()
+    const uuid = new UUIDT().toString()
+    const sessionId = new UUIDT().toString()
+    const veryLongString = generateVeryLongString()
+    await capture({
```

Comment:
fwiw I think you'll want to specify the topic to be the `session_recording_events` topic here. For backwards compat. we're forwarding these events from the `events_plugin_ingestion` topic but we should remove this.

Reply:
🤔